addon/services/sticky-manager.js

```diff
@@ -0,0 +1,245 @@
+        return {
+          sticky: false,
+          top: 0,
+          left: 0,
+          width: 'auto',
+        };
+      }
+
+      let stickyOffset = offset
+        ? this.getOffset(stickyNode) + parseInt(tokens.spacingLoose, 10)
+        : this.getOffset(stickyNode);
+
+      let scrollPosition = scrollTop + stickyOffset;
+      let placeHolderNodeCurrentTop =
+        getRectForNode(placeHolderNode).top - containerTop + scrollTop;
+      let top = containerTop + stickyOffset;
+      let width = getRectForNode(placeHolderNode).width;
+      let left = getRectForNode(placeHolderNode).left;
+
+      let sticky;
+
+      if (boundingElement == null) {
+        sticky = scrollPosition >= placeHolderNodeCurrentTop;
+      } else {
+        let stickyItemHeight = getRectForNode(stickyNode).height;
+        let stickyItemBottomPosition =
+          getRectForNode(boundingElement).bottom -
+          stickyItemHeight +
+          scrollTop -
+          containerTop;
+
+        sticky =
+          scrollPosition >= placeHolderNodeCurrentTop &&
+          scrollPosition < stickyItemBottomPosition;
+      }
+
+      return {
+        sticky,
+        top,
+        left,
+        width,
+      };
+    },
+
+    updateStuckItems(item, sticky) {
+      let { stickyNode } = item;
+      if (sticky && !this.isNodeStuck(stickyNode)) {
+        this.addStuckItem(item);
+      } else if (!sticky && this.isNodeStuck(stickyNode)) {
+        this.removeStuckItem(item);
+      }
+    },
+
+    addStuckItem(stickyItem) {
+      this.get('stuckItems').push(stickyItem);
+    },
+
+    removeStuckItem(stickyItem) {
+      let stuckItems = this.get('stuckItems');
+      let { stickyNode: nodeToRemove } = stickyItem;
+      let nodeIndex = stuckItems.findIndex(
+        ({ stickyNode }) => nodeToRemove === stickyNode
+      );
+      stuckItems.splice(nodeIndex, 1);
+    },
+
+    getOffset(node) {
+      let stuckItems = this.get('stuckItems');
+      let stuckItemsLength = stuckItems.get('length');
+      if (stuckItemsLength === 0) {
+        return 0;
+      }
+
+      let offset = 0;
+      let count = 0;
+      let stuckNodesLength = stuckItemsLength;
+      let nodeRect = getRectForNode(node);
+
+      while (count < stuckNodesLength) {
+        let stuckNode = stuckItems[count].stickyNode;
+        if (stuckNode !== node) {
+          let stuckNodeRect = getRectForNode(stuckNode);
+          if (!horizontallyOverlaps(nodeRect, stuckNodeRect)) {
+            offset += getRectForNode(stuckNode).height;
+          }
+        } else {
+          break;
+        }
+        count++;
+      }
+
+      return offset;
+    },
+
+    isNodeStuck(node) {
+      let nodeFound = this.get('stuckItems').findIndex(
+        ({ stickyNode }) => node === stickyNode
+      );
+
+      return nodeFound >= 0;
+    },
+
+    init() {
+      this._super(...arguments);
+
+      let container = this.get('container') || document;
```

Comment:
The original `StickyManager` React code doesn't default this to `document`, but the React `AppProvider` _does_ [set it to `document`](https://github.com/Shopify/polaris/blob/master/src/components/AppProvider/AppProvider.tsx#L48) if `document` exists.

Reply:
can you add this comment in code pls?